### PR TITLE
fix(ColorTransferFunction): applyColorMap calls modified in more cases

### PR DESCRIPTION
**Pull request by PaulHax**
### Context
Similar shaped color maps passed to `ColorTransferFunction.applyColorMap()` did not always trigger publicAPI.modified()

### Results

More exhaustive change checking in `applyColorMap`

### Changes

- [ ] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist

- [ ] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [ ] Run `npm run reformat` to have correctly formatted code

### Testing

- [x] This change adds or fixes unit tests 
- [x] Tested environment:
  - **vtk.js**: 25.9.2
  - **OS**: LInux
  - **Browser**: Chrome 105

### CODE REVIEW

**Comment by floryst** [comment]
LGTM!

**Review by finetjul — COMMENTED** [review]
Peformance-wise, that's too bad you "stringify" many objects when maybe the first `colorSpace` new value would trigger `modified`.

LGTM otherwise.

**Comment by PaulHax** [comment]
> Peformance-wise, that's too bad you "stringify" many objects when maybe the first colorSpace new value would trigger modified.

Challenge accepted =)

File: Sources/Rendering/Core/ColorTransferFunction/index.js [hunk 1]
```diff
@@ -1161,12 +1162,18 @@ function vtkColorTransferFunction(publicAPI, model) {
         model.colorSpace = ColorSpace.RGB;
       }
     }
+    let isModfied = oldColorSpace !== JSON.stringify(model.colorSpace);
+
+    const oldNanColor = JSON.stringify(model.nanColor);
```
Comment (finetjul) [inline Sources/Rendering/Core/ColorTransferFunction/index.js:1167]: const oldNanColor = isModfied || JSON.stringify(model.nanColor);

File: Sources/Rendering/Core/ColorTransferFunction/index.js [hunk 2]
```diff
@@ -1161,12 +1162,18 @@ function vtkColorTransferFunction(publicAPI, model) {
         model.colorSpace = ColorSpace.RGB;
       }
     }
+    let isModfied = oldColorSpace !== JSON.stringify(model.colorSpace);
```
Comment (finetjul) [inline Sources/Rendering/Core/ColorTransferFunction/index.js:1165]: isModfied -> isModified

**Review by finetjul — APPROVED** [review]

File: Sources/Rendering/Core/ColorTransferFunction/test/testColorTransferFunction.js [hunk 3]
```diff
@@ -159,3 +159,61 @@ test('Test discretized color transfer function', (t) => {
...
+    RGBPoints: [0, 0, 0, 0, 1, 0, 0, 0],
+  };
+  const colorMapB = {
+    ColorSpace: 'RGB',
+    RGBPoints: [0, 0, 0, 0, 1, 1, 1, 1],
+  };
+
+  ctf.applyColorMap(colorMapA);
+
+  let isModfied = false;
```
Comment (finetjul) [inline Sources/Rendering/Core/ColorTransferFunction/test/testColorTransferFunction.js:177]: isModfied -> isModified

Comment (PaulHax) [reply Sources/Rendering/Core/ColorTransferFunction/test/testColorTransferFunction.js]: 🙏

**Review by finetjul — APPROVED** [review]

**Pull request merged**